### ISSUE TITLE
Guard is now a new profession with trades, Dogs can spawn in villages attracted to invidiual owners, villager rings bell when in someone in danger.

### ISSUE THREAD

**Issue by babacon88** [seymourimadeit/guardvillager-suggestions#121]
_Guards are new a villager with specialized profession and its level system, which grants them inventory, ability to share food, breed, need a bed to sleep
 _They will either need kills to gain xp, or trades. They can sell you fleshes and bones from slain zombie, while buy armors, steel, weapons and food. a guard post act as a professional table should also be added.
_Gaining traits related to their trades.
_Guards ranged weapons  can have a trait to seek and snipe down far away foes while maintaining undetectable distance, they may not give chases to far away hostiles, require a direct line of sight.
_gain traits to use crushing blows, sweep attack without friendly fire, spawn arrows to maintain distance, will not do a full 
bow draw if the target is at critical health.
_they no longer hold 2 weapons at one but switch between them and only hold 2 within combat. They hold ranged weapon with 2 hands, and switch to swords and shields when having to melee
_Disable ranged friendly fire, or make it a trait for guards with higher level.
_Low health guards will attempt to stall hostiles with shields to stay alive longer, they hold shields indefinitely when being targeted in critial health.
_Barracks or guard towers containing their workbenches is now able to spawn along with villagers, or randomly as a structure.
_Guards and occasionally some villagers, has a chance to have isomania/night own trait, they sleep during day and be on duty at night when village need him the most.
_Dogs are spawn in Villages, attached to invidual owners, whom they often check on and follow occsionally.
_when a member of village reaches half heath/killed by hostile. villagers will seek to ring the bell. Making all guards become extremely active and seek out hostiles.
_non professionalized villagers may attempt to pick up weapons and armors from their fallen guard brothers, thus keeping tradition alive.
_Guards are now just a villager with a new, specialized job, so they need bed, have inventory, take and share food, breed,... Since their is no workbenches for now, a guard may attempt to train a youngling to become a guard, passing the sword down to generations
_New uses for shields: They can be deployed, stand to the ground, carry behind the back blocking projectiles pavise style, Can be a new AI combat style for guards, who will attemp to turn their back/deployshields against skeletons, if they cant reach their shield after being deplyed, it would spawn back into their inventory. Adding spears for proper phalanx/ pike and shoot formation
_golems are still really weak and not really worth of 9x4 irons in vanilla for lack of surviabilty, smiths should be able to upgrade/spawn golems with more trait to deal with their fatal weakness:
+sweeping+crushing combo to deal with deal fatal weakness: being grouped on by zombies, friendly fire disabled of course
+Gaining immunity to arrow, tridents,...: given atleast 80% change of blocking arrows completely or take less damges from arrows. Can be bypassed with piercing crossbow enchant. most of the times golem just get shot to death by village guards 
+additional traits are increase health or getting a ranged attack,...
+they should have immunity to fire.
That's all my limited mind can think of, thank you for getting to read this far. I hope someone my ideas catch your interests in the next update.
You should also check out the mod: Village's artifacts and do a compatility with them.

**Comment by omeggga** [comment]
I had the idea of guards gaining emeralds for every kill or assist they get and using those emeralds to trade with villagers. I personally think that would be a more natural progression system. Other than that most of these ideas are pretty solid.

**Comment by babacon88** [comment]
Yes, years ago I played with a mod: Dragon: Ice and fire that added kings, lords, village elders that give you quests to kills certain amount of monster within their boundary
shame they no longer works on the mod

**Comment by omeggga** [comment]
That's not what I was saying though, I was saying every time a guard kills something they get an emerald, and the guards trade for armor and weapons with other villagers using those emeralds.

**Comment by babacon88** [comment]
Kinda cruel to have a guard pays for his own gears to protect the village
yeah thats something villagers would do

**Comment by omeggga** [comment]
> Kinda cruel to have a guard pays for his own gears to protect the village yeah thats something villagers would do

I mean yeah in a sense but tbh mercs in the middle ages were given standard gear and expected to deal with their own gear.
Besides, villagers expect you to periodically trade with them, you, the guy who improves their village. If you don't you lose their discounts.
